weatherman.py

```diff
@@ -0,0 +1,164 @@
+import os
+import argparse
+from datetime import datetime
+
+
+RED = "\033[31m"
+BLUE = "\033[34m"
+RESET = "\033[0m"
+
+
+class WeatherReading:
+    def __init__(self, date, max_temp, min_temp, max_humidity, mean_humidity):
+        self.date = date
+        self.max_temp = max_temp
+        self.min_temp = min_temp
+        self.max_humidity = max_humidity
+        self.mean_humidity = mean_humidity
+
+    def __repr__(self):
+        return (f"WeatherReading(date={self.date}, max_temp={self.max_temp}, "
+                f"min_temp={self.min_temp}, max_humidity={self.max_humidity}, "
+                f"mean_humidity={self.mean_humidity})")
+
+
+def parse_weather_file(file_path):
+    readings = []
+    try:
+        with open(file_path, 'r') as file:
+            lines = file.readlines()[1:]  
+            for line in lines:
+                parts = line.strip().split(',')
+                if len(parts) < 9 or '' in (parts[1], parts[3], parts[7], parts[8]):
+                    continue  
+                date = parts[0]
+                max_temp = int(parts[1])
+                min_temp = int(parts[3])
+                max_humidity = int(parts[7])
+                mean_humidity = int(parts[8])
```

Comment:
Why we are explicitly parsing the data to int? Temperature can be in float as well, no?

Reply:
Yes, temperature can be represented as float. In this particular scenario, all the values we were using were integers, which is why I used integers. However, for the sake of generalization, I have switched to using floats. This is because no data is lost when converting an integer to a float, but the same cannot be said for the reverse conversion.